wave_vision/include/wave/vision/matcher/descriptor_matcher.hpp

```diff
@@ -0,0 +1,71 @@
+/**
+ * @file
+ * Base class from which descriptor matchers can be derived.
+ * @ingroup vision
+ */
+#ifndef WAVE_VISION_DESCRIPTOR_MATCHER_HPP
+#define WAVE_VISION_DESCRIPTOR_MATCHER_HPP
+
+/** C++ Headers */
+#include <exception>
+
+/** Third Party Headers */
+#include <opencv2/opencv.hpp>
+
+/** Libwave Headers */
+#include "wave/utils/utils.hpp"
+
+namespace wave {
+/** @addtogroup vision
+ *  @{ */
+
+/** Representation of a generic descriptor matcher.
+ *
+ *  Internally, this class, and all derived classes are wrapping various
+ *  descriptor matchers implemented in OpenCV. Further reference on descriptor
+ *  matchers can be found [here][opencv_descriptor_matchers].
+ *
+ *  [opencv_descriptor_matchers]:
+ *  http://docs.opencv.org/trunk/db/d39/classcv_1_1DescriptorMatcher.html
+ */
+class DescriptorMatcher {
+ public:
+    virtual ~DescriptorMatcher() = default;
```

Comment:
No. Actually, because I've also deleted the destructor and the showMatches method, there's really zero point of inheritance here (the only method was the overridden matchDescriptors method). So I've deleted the entire descriptorMatcher class.